web/src/components/dynamic-loader/dynamic-loader.tsx

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as React from 'react';
+import * as reactRouterDom from 'react-router-dom';
+
+/**
+ * useHistory is deprecated and not available anymore in react-router 6+
+ * This function allow retro compatibility between useHistory and useNavigate
+ * by switching from one to the other
+ */
+type NavFunc = (to: string, opts?: any) => void;
+let navFunc: NavFunc | null = null;
+
+export const setNavFunction = (f: NavFunc) => {
+  navFunc = f;
+};
+
+export const navigate = (to: string, opts?: any) => {
+  if (navFunc) {
+    navFunc(to, opts);
+  } else {
+    console.error('navigate error; navFunc is not initialized', navFunc);
+  }
+};
+
+export const DynamicLoader: React.FC<{}> = ({ children }) => {
+  const genericReactRouterDom = reactRouterDom as any;
+  if (genericReactRouterDom['useNavigate']) {
+    console.log('loading nav function from react-router useNavigate');
+    setNavFunction(genericReactRouterDom['useNavigate']());
+  } else if (genericReactRouterDom['useHistory']) {
+    console.log('loading nav function from react-router useHistory');
+    setNavFunction(genericReactRouterDom['useHistory']().push);
+  } else {
+    console.error("can't load nav function from react-router");
+  }
```

Comment:
Just tested on my 4.13 cluster and it uses `useHistory`. I guess I should spin up a 4.14 to test useNavigate?

Reply:
`useNavigate` will be used only when console plugin team will finish the migration introducing v6 API.
- prior to 4.14 we rely on v5 api
- ocp 4.14 will introduce `react-router-dom-v5-compat` with v6 capabilities using this compat lib. Since we don't import it it will still rely on v5 api
- a later version will introduce v6 api, that's when we'll switch to `useNavigate`

Reply:
https://redhat-internal.slack.com/archives/C011BL0FEKZ/p1687957240461039